### PR TITLE
caddyhttp: Address some Go 1.20 features

### PR DESCRIPTION
**Pull request by mholt**
See #5288.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -40,6 +40,18 @@ import (
...
+	// headaches related to establishing TLS connections.
+	// "Package fallback embeds a set of fallback X.509 trusted
+	// roots in the application by automatically invoking
+	// x509.SetFallbackRoots. This allows the application to
+	// work correctly even if the operating system does not
+	// provide a verifier or system roots pool. ... It's
+	// recommended that only binaries, and not libraries,
+	// import this package. This package must be kept up to
+	// date for security and compatibility reasons."
+	_ "golang.org/x/crypto/x509roots/fallback"
```
Comment (francislavoie) [inline cmd/main.go:54]: This was my idea... But does it increase the binary size significantly?

Comment (mholt) [reply cmd/main.go:46]: The difference in the size of the artifacts is .2-.3 MB (200-300 KB).

Comment (francislavoie) [reply cmd/main.go:46]: Oh that's not bad at all

**Comment by francislavoie** [comment]
Gah. Linter is running into this issue: https://github.com/daixiang0/gci/issues/76 both `goimports` and `gci` are fighting eachother on the whitespace before the block comment.

As a workaround, I'm trying to move this import to a file of its own, with the comment _above_ the `import` so it shouldn't be touched by either tool. I think that should work, right?

**Comment by mholt** [comment]
Ah, I wondered... Because I manually put it at the very next line but goimports kept adding a new line when I saved. I figured that must be the wrong place, but I guess not.

It looks like the issue upstream in gci is closed? So maybe there's a resolution.

**Comment by francislavoie** [comment]
> It looks like the issue upstream in gci is closed? So maybe there's a resolution.

Nope, it was closed prematurely

**Review by francislavoie — APPROVED** [review]

**Pull request merged**